src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala

```diff
@@ -229,6 +229,30 @@ class FileRepositorySpec extends TestContainerUtils with ScalaFutures with Match
       files.size shouldBe 1
   }
 
+  "getFiles" should "return files and file metadata where 'selectedFileIds' filter applied" in withContainers {
+    case container: PostgreSQLContainer =>
+      val db = container.database
+      val utils = TestUtils(db)
+      val fileRepository = new FileRepository(db)
+      val consignmentId = UUID.fromString("c6f78fef-704a-46a8-82c0-afa465199e66")
+      setUpFilesAndDirectories(consignmentId, utils)
+
+      val files = fileRepository.getFiles(consignmentId, FileFilters(selectedFileIds = Some(List(fileOneId)))).futureValue
+      files.forall(_._1.fileid == fileOneId) shouldBe true
+  }
+
+  "getFiles" should "return files and file metadata where and 'file' type and 'selectedFileIds' filter applied" in withContainers {
```

Comment:
Would reword the test slightly to: `... return files and file metadata for selected files where 'file' type and 'selectedFileIds' filters are applied`